app/controllers/pages_controller.rb

```diff
@@ -8,11 +8,13 @@ def index
       @posts = current_user.posts
         .primary
         .includes(:tags)
-        .order(created_at: :desc)
-        .paginate(page: params[:page], per_page: 10)
+
+      @citations = current_user.posts.generated
 
       @tags = current_user.tags
 
+      @user = current_user
```

Comment:
what is the purpose of this?

Reply:
user id is needed in `generateUrl()` in `PostContainer` 

```
const generateUrl = () => {
  const type = props.tag ? 'tags' : 'users'
  const id = props.tag ? props.tag.id : props.user.id
  const tab = tabState === '1' ? 'posts' : 'citations'
  return '/' + type + '/' + id + '/paginated_' + tab + '/' + page
}
```

`@user` in `dashboard.html.slim` is passed in as a prop to `PostsContainer`

```
= react_component "PostsContainer",
user: @user,
postsCount: @posts.count,
citationsCount: @citations.count,
isLibrary: true
```

Reply:
Another option is to use our redux store to get access to the user, but to the best of my understanding `store.getState().currentUser` returns an empty object unless `PostEditor` component is rendered. 

Speaking as someone who does not understand redux store.

Reply:
Removed `@user = current_user`. TIL we have access to `currentUser` everywhere.